### PR TITLE
build_image: Allow a two-part version value leaving out FLATCAR_PATCH

### PR DESCRIPTION
**Pull request by pothos**
When a version scheme with x.y is detected, don't include the
FLATCAR_PATCH field in the version.txt file.

# How to use

Build an LTS image with a two-part version scheme.

# Testing done

Image built, checked the contents of /etc/os-release /usr/share/flatcar/release and the version.txt file.

### CODE REVIEW

File: build_image [hunk 1]
```diff
@@ -196,7 +196,11 @@ split_ver "${FLATCAR_VERSION_ID}" SPLIT
 tee "${BUILD_DIR}/version.txt" <<EOF
 FLATCAR_BUILD=${SPLIT[0]}
 FLATCAR_BRANCH=${SPLIT[1]}
-FLATCAR_PATCH=${SPLIT[2]}
+EOF
+if [[ ${#SPLIT[@]} -eq 3 ]]; then
```
Comment (vbatts) [inline build_image:200]: probably could end up in common.sh

**Comment by vbatts** [comment]
Seems like a good starting point. I'm sure we'll find various places that expect xx.yy.zz, so this will be iterative

File: build_image (hunk 1 above)
Comment (pothos) [reply build_image:200]: The whole file creation part here? The version.txt is only created once for an image.

Comment (vbatts) [reply build_image:200]: sorry, was thinking about the version comparisons.

**Review by marga-kinvolk — APPROVED** [review]
LGTM assuming tests pass :)

**Pull request closed without merging**